code/CMSBatchActionHandler.php

```diff
@@ -125,11 +125,8 @@ public function Link($action = null)
 
     /**
      * Invoke a batch action
-     *
-     * @param HTTPRequest $request
-     * @return HTTPResponse
      */
-    public function handleBatchAction($request)
+    public function handleBatchAction(HTTPRequest $request): HTTPResponse
```

Comment:
Could `$actionHandler->run()` in custom code return something other than `HTTPResponse`? If so, this return type is too restrictive.

Reply:
Yes it could. I've reverted the changes to this method

Reply:
Possibly it could, though we're updating the signature on the abstract method `CMSBatchAction::run()` to include the `HTTPResponse` return type i.e. `abstract public function run(SS_List $objs): HTTPResponse;`

Because we're updating this, any project that have custom code (which honestly seems pretty unlikely) will immediately see the the return type on their `run()` method needs to be updated.